client/my-sites/patterns/components/pattern-preview/index.tsx

```diff
@@ -271,6 +271,29 @@ function PatternPreviewFragment( {
 		};
 	} );
 
+	// This fetches forms and adds a listener that disables submission
+	useEffect( () => {
+		const iframe = ref.current?.querySelector( 'iframe' );
+
+		if ( ! iframe?.contentDocument ) {
+			return;
+		}
+
+		const forms = Array.from( iframe.contentDocument.querySelectorAll( 'form' ) );
+
+		if ( ! forms.length ) {
+			return;
+		}
```

Comment:
```suggestion
		const forms = iframe.contentDocument.querySelectorAll( 'form' );
```
Somewhat nitpicky, but we don't actually need to cast the `NodeList` to an array (`forEach` is supported by `NodeList`, too). We also don't really need to return early if the list is empty.

Reply:
> We also don't really need to return early if the list is empty.

The advantage of returning early is, we don't need to create nor allocate unneeded resources.  Also even though the list is empty, a function is invoked, nevertheless.